runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -535,24 +532,28 @@ xlpSubOptionsParser(J9JavaVM *vm, IDATA xlpIndex, XlpError *xlpError, UDATA *req
 	}
 
 #if defined(J9ZOS390)
-	/*
-	 *  [non]pageable
-	 *  - this option must be specified for Z platforms
-	 */
-	if ((0 == pageableHowMany) && (0 == nonPageableHowMany)) {
-		/* error: [non]pageable not found */
-		xlpErrorState = XLP_INCOMPLETE_OPTION;
-		xlpError->xlpOptionErrorString = "-Xlp:objectheap:";
-		xlpError->xlpMissingOptionString = "[non]pageable";
-		return xlpErrorState;
-	}
-
-	if (pageableOptionNumber > nonPageableOptionNumber) {
+	if (nonPageableOptionNumber > pageableOptionNumber) {
+		/* nonpageable is most right */
+		*requestedPageFlags = J9PORT_VMEM_PAGE_FLAG_FIXED;
+	} else if (pageableOptionNumber != 0) {
 		/* pageable is most right */
 		*requestedPageFlags = J9PORT_VMEM_PAGE_FLAG_PAGEABLE;
 	} else {
-		/* nonpageable is most right */
-		*requestedPageFlags = J9PORT_VMEM_PAGE_FLAG_FIXED;
+		/* [non]pageable not passed. By default use pageable if both are sizes are available */
+		UDATA *pageSizes = j9vmem_supported_page_sizes();
+		UDATA *pageFlags = j9vmem_supported_page_flags();
+		for (UDATA pageIndex = 0; 0 != pageSizes[pageIndex]; ++pageIndex) {
+			if (pageSizes[pageIndex] == (UDATA)requestedPageSize) {
+
+				if (pageFlags[pageIndex] == J9PORT_VMEM_PAGE_FLAG_PAGEABLE) {
+					*requestedPageFlags = J9PORT_VMEM_PAGE_FLAG_PAGEABLE;
+					break;
+				}
+				else {
+					*requestedPageFlags = J9PORT_VMEM_PAGE_FLAG_FIXED;
+				}
+			}
+		}
```

Comment:
This change looks in wrong place. Parser does context free check. This code uses context (pages available on OS).
Looks like it should be somewhere in `j9vmem_find_valid_page_size()`

Reply:
So directly implemented into
https://github.com/eclipse/omr/blob/master/port/zos390/omrvmem.c#L1331

Reply:
> So directly implemented into
> https://github.com/eclipse/omr/blob/8b719f55ba3d8b8051b40b6dc4b38a1caa9afe5e/port/zos390/omrvmem.c#L1331

I believe this is the question. 
I think you should take a look. It supports `OMRPORT_VMEM_PAGE_FLAG_NOT_USED` so there is a mechanism to handle unspecified `[non]pageable` there. However it seems does it differently and favour FIXED in this case as a legacy support.
I believe we need a discussion between ZOS - involved specialists about this.

Reply:
> I believe we need a discussion between ZOS - involved specialists about this proposed behaviour change (including motivation)
@joransiu Has already expressed that 1M pagerable large pages are preferred. If we don't have any obvious drawbacks I don't see why this can't be default behaviour. I think it adds value that the the command line option can be symmetrical along all architectures.

It's obvious on the codecache side that we don't have a choice of using nonpageable in the first place and mandatory requiring an additional option doesn't make sense.